packages/web/src/components/create-playlist/PlaylistForm.tsx

```diff
@@ -0,0 +1,258 @@
+
+import { Button, ButtonType, IconCheck } from '@audius/stems'
+import { debounce } from 'lodash'
+
+import { Collection, CollectionMetadata } from 'common/models/Collection'
+import { SquareSizes } from 'common/models/ImageSizes'
+import { Nullable, DeepNullable } from 'common/utils/typeUtils'
+import Input from 'components/data-entry/Input'
+import TextArea from 'components/data-entry/TextArea'
+import UploadArtwork from 'components/upload/UploadArtwork'
+import { useCollectionCoverArt } from 'hooks/useCollectionCoverArt'
+import * as schemas from 'schemas'
+import { resizeImage } from 'utils/imageProcessingUtil'
+
+import styles from './PlaylistForm.module.css'
+
+const messages = {
+  editPlaylistButtonText: 'Save Changes',
+  cancelButtonText: 'Cancel',
+  deletePlaylistButtonText: 'Delete Playlist',
+  deleteAlbumButtonText: 'Delete Album',
+  createPlaylistButtonText: 'Create Playlist'
+}
+
+type PlaylistFormFields = Partial<Collection> & {
+  artwork: {
+    file: Blob
+    url: string
+    source: 'unsplash' | 'original'
+    error?: string
+  }
+  is_current: boolean
+  mood: Nullable<string>
+  created_at: Nullable<string>
+  tags: Nullable<string>
+  genre: Nullable<string>
+  isAlbum: boolean
+} & DeepNullable<
+    Pick<
+      CollectionMetadata,
+      | 'is_private'
+      | 'updated_at'
+      | 'cover_art'
+      | 'cover_art_sizes'
+      | 'playlist_name'
+      | 'playlist_owner_id'
+      | 'save_count'
+      | 'upc'
+      | 'description'
+    >
+  >
+
+type PlaylistFormProps = {
+  metadata?: Nullable<Collection>
+  isAlbum?: boolean
+  onOpenArtworkPopup?: () => void
+  onCloseArtworkPopup?: () => void
+  isEditMode?: boolean
+  /** Only applies to edit mode */
+  onDelete?: () => void
+  /** Only applies to edit mode */
+  onCancel?: () => void
+  onSave: (formFields: PlaylistFormFields) => void
+}
+
+const EditPlaylistActions = ({
+  isAlbum,
+  onDelete,
+  onCancel,
+  onSave
+}: {
+  isAlbum: boolean
+  onDelete?: () => void
+  onCancel?: () => void
+  onSave: () => void
+}) => {
+  return (
+    <div className={styles.editPlaylistActionsContainer}>
+      <div>
+        <Button
+          text={
+            isAlbum
+              ? messages.deleteAlbumButtonText
+              : messages.deletePlaylistButtonText
+          }
+          type={ButtonType.SECONDARY}
+          onClick={onDelete}
+          className={styles.deleteButton}
+          textClassName={styles.deleteButtonText}
+        />
+      </div>
+      <div>
+        <Button
+          text={messages.cancelButtonText}
+          type={ButtonType.SECONDARY}
+          className={styles.cancelButton}
+          textClassName={styles.cancelButtonText}
+          onClick={onCancel}
+        />
+        <Button
+          className={styles.saveChangesButton}
+          text={messages.editPlaylistButtonText}
+          type={ButtonType.SECONDARY}
+          onClick={debounce(onSave, 500, { leading: true })}
```

Comment:
oh this is interesting, is this to prevent extra clicks?

Reply:
I think this was a carry-over from the existing modal (js) but maybe instead we could disable the button immediately on click in a callback that only then calls `onSave`?

Reply:
yeah it was from the old modal. im not really sure why we had it, maybe @raymondjacobson would know? if it was just to prevent extra clicks then agree that disabling is better

Reply:
I'm sure I was the one who rote this, haha. It was definitely to prevent extra clicks.